tsdb/engine/tsm1/engine.go

```diff
@@ -228,7 +258,8 @@ func (e *DevEngine) Begin(writable bool) (tsdb.Tx, error) {
 
 func (e *DevEngine) WriteTo(w io.Writer) (n int64, err error) { panic("not implemented") }
 
-func (e *DevEngine) writeSnapshot() error {
+// WriteSnapshot will snapshot the cache and write a new TSM file with its contents, releasing the snapshot when done.
+func (e *DevEngine) WriteSnapshot(async bool) error {
```

Comment:
Multiple snapshots can be run at the same time. This is not currently true for compaction (and I'm not sure we ever need to worry about it for compaction).

Snapshots can quickly become a bottleneck because the WAL can stream data to disk very quickly. Having multiple snapshots run at the same time helps alleviate this.

I'm not sure what having a long running goroutine dedicated to snapshots gives us. How's that different than just kicking off goroutine as needed?